### PR TITLE
fix(plugins): remove edit plugin command.

### PR DESCRIPTION
**Pull request by edgarulg**
Currently, the command `hal plugins edit` return null and is not working, the `edit command` did not support updating the config for the plugin.  So let's just edit by hand for now.

### CODE REVIEW

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/plugins/EditPluginCommand.java [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-    for (String pluginExtensionName : extensions.split(",")) {
-      plugin
-          .getExtensions()
-          .putIfAbsent(pluginExtensionName, new PluginExtension().setId(pluginExtensionName));
-    }
-
-    new OperationHandler<Void>()
-        .setFailureMesssage("Failed to edit plugin " + plugin.getId() + ".")
-        .setSuccessMessage("Successfully edited plugin " + plugin.getId() + ".")
-        .setOperation(Daemon.setPlugin(currentDeployment, plugin.getId(), !noValidate, plugin))
```
Comment (danielpeach) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/plugins/EditPluginCommand.java:62]: Does anything else use `Daemon.setPlugin`? If not, let's rip it out and whatever else is exercised when you call it (there should be a corresponding controller method).

File: halyard-cli/src/test/groovy/com/netflix/spinnaker/halyard/cli/command/v1/CommandTreeSpec.groovy [hunk 2]
```diff
@@ -38,10 +38,9 @@ import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.x509.X
 import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authz.AuthzCommand
 import com.netflix.spinnaker.halyard.cli.command.v1.config.security.ui.UiSecurityCommand
 import com.netflix.spinnaker.halyard.cli.command.v1.config.stats.StatsCommand
-import com.netflix.spinnaker.halyard.cli.command.v1.config.stats.StatsEnableDisableCommandBuilder
 import com.netflix.spinnaker.halyard.cli.command.v1.plugins.AddPluginCommand
 import com.netflix.spinnaker.halyard.cli.command.v1.plugins.DeletePluginCommand
-import com.netflix.spinnaker.halyard.cli.command.v1.plugins.EditPluginCommand
+
```
Comment (danielpeach) [inline halyard-cli/src/test/groovy/com/netflix/spinnaker/halyard/cli/command/v1/CommandTreeSpec.groovy:43]: Please remove this line.

**Comment by danielpeach** [comment]
Just want to point out the background on this - this code never worked, and we don't want to invest the time in supporting it.

File: halyard-cli/src/test/groovy/com/netflix/spinnaker/halyard/cli/command/v1/CommandTreeSpec.groovy (hunk 2 above)
Comment (edgarulg) [reply halyard-cli/src/test/groovy/com/netflix/spinnaker/halyard/cli/command/v1/CommandTreeSpec.groovy]: i removed it

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/plugins/EditPluginCommand.java (hunk 1 above)
Comment (edgarulg) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/plugins/EditPluginCommand.java:62]: The `Daemon.setPlugin` is not used by anything else, the only place where it is use is in the EditPluginCommand that I already removed

Comment (edgarulg) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/plugins/EditPluginCommand.java:62]: I removed the unused code for the `Daemon.setPlugin` on the service, also I removed the endpoint on the Web project that edit a plugin via Rest API.

**Comment by maggieneterval** [comment]
@edgarulg @danielpeach CI for github.spinnaker.io is [failing at HEAD](https://github.com/spinnaker/spinnaker.github.io/runs/785406488) because there's a lingering reference to the now-non-existent edit plugin command; mind updating the plugin docs to remove that reference?

**Comment by edgarulg** [comment]
Hello @maggieneterval my apologize for that, I already remove the documentation on this PR: https://github.com/spinnaker/spinnaker.github.io/pull/1885 Thanks for taking a look

**Pull request merged**